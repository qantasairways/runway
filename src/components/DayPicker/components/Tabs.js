import React from 'react';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import Triangle from './TriangleSvg';
import {
  colours,
  mq,
  layout,
  fontSize,
  fontWeight,
  fontFamily
} from '../../../theme/airways';
import {
  CSS_PSEUDO_AFTER,
  CSS_SELECTOR_FOCUS
  // CSS_SELECTOR_HOVER
} from '../../../constants/css';

export function calendarHeatherCntStyle() {
  return css({
    position: 'relative',
    width: '100%',
    backgroundColor: colours.darkerGrey,
    display: 'flex'
  });
}

const tabStyles = {
  fontFamily: fontFamily.main,
  position: 'relative',
  // cursor: 'pointer',
  width: '50%',
  height: '64px',
  display: 'flex',
  alignItems: 'center',
  [CSS_SELECTOR_FOCUS]: {
    outline: 'none'
  },
  [mq.medium]: {
    height: '80px'
  },
  [mq.large]: {
    justifyContent: 'center'
  }
};

export function endTabStyle({ isSelectingStartDate }) {
  return css(tabStyles, {
    label: 'runway-calendar-header_endTab_style',
    color: isSelectingStartDate ? colours.white : colours.darkerGrey,
    border: isSelectingStartDate ? 'none' : `solid 5px ${colours.highlights}`,
    borderLeft: 'none',
    backgroundColor: isSelectingStartDate ? colours.darkerGrey : colours.white,
    textAlign: 'right',
    float: 'right',
    justifyContent: 'flex-end',
    padding: `0 ${layout.gutter}`,
    // [CSS_SELECTOR_HOVER]: {
    //   backgroundColor: isSelectingStartDate ? colours.mediumGrey : colours.white
    // },
    [mq.medium]: {
      justifyContent: 'flex-start'
    }
  });
}

export function startTabStyles({ isSelectingStartDate }) {
  return css(tabStyles, {
    label: 'runway-calendar-header_startTab_style',
    padding: `0 ${layout.gutter}`,
    border: isSelectingStartDate
      ? `solid 5px ${colours.highlights}`
      : `solid 5px ${colours.darkerGrey}`,
    borderRight: '0',
    backgroundColor: isSelectingStartDate ? colours.white : colours.darkerGrey,
    color: isSelectingStartDate ? colours.darkerGrey : colours.white,
    boxSizing: 'border-box',
    justifyContent: 'space-between',
    textAlign: 'left',
    // [CSS_SELECTOR_HOVER]: {
    //   backgroundColor: isSelectingStartDate ? colours.white : colours.mediumGrey
    // },
    [CSS_PSEUDO_AFTER]: {
      content: "''",
      zIndex: '5',
      position: 'absolute',
      right: isSelectingStartDate ? '-14px' : '-1px',
      borderTop: `32px solid ${colours.transparent}`,
      borderLeft: isSelectingStartDate ? `15px solid ${colours.white}` : '0',
      borderBottom: `32px solid ${colours.transparent}`,
      borderRight: isSelectingStartDate ? '0' : `19px solid ${colours.white}`,
      top: '-5px',
      [mq.medium]: {
        right: isSelectingStartDate ? '-17px' : '-1px',
        borderLeft: isSelectingStartDate ? `19px solid ${colours.white}` : '0',
        borderTop: `42px solid ${colours.transparent}`,
        borderBottom: `42px solid ${colours.transparent}`,
        top: '-4px'
      }
    },
    [mq.medium]: {
      justifyContent: 'flex-end'
    }
  });
}

export function textStyles({ isPlaceholder }) {
  return css({
    label: 'runway-calendar-header_text_style',
    fontSize: fontSize.body,
    lineHeight: isPlaceholder ? 1.11 : 1.25,
    fontWeight: fontWeight.regular,
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    [mq.medium]: {
      fontSize: fontSize.medium,
      maxWidth: '500px',
      justifyContent: 'center'
    },
    [mq.large]: {
      flexDirection: 'row'
    }
  });
}

export function iconStyles({ isSelectingStartDate }) {
  return css({
    width: '28px',
    height: '28px',
    zIndex: '10',
    position: 'absolute',
    right: isSelectingStartDate ? '-7px' : '-19px',
    transform: isSelectingStartDate ? 'rotate(0deg)' : 'rotate(180deg)',
    [mq.small]: {
      width: '32px',
      height: '32px',
      right: isSelectingStartDate ? '-7px' : '-24px'
    },
    [mq.medium]: {
      width: '40px',
      height: '40px',
      right: isSelectingStartDate ? '-3px' : '-34px'
    }
  });
}

export function TriangleSvgStyles({ isSelectingStartDate }) {
  return css({
    zIndex: '10',
    position: 'absolute',
    right: isSelectingStartDate ? '-48px' : '-12px',
    top: isSelectingStartDate ? '-5px' : '-7px',
    transform: isSelectingStartDate ? 'rotate(0)' : 'rotate(-180deg)',
    [mq.medium]: {
      height: isSelectingStartDate ? '84px' : '83px',
      right: isSelectingStartDate ? '-53px' : '-7px',
      top: isSelectingStartDate ? '-5px' : '-7px'
    }
  });
}

const renderTabContent = (date, label, placeholder, isSelected) => {
  if (date) {
    return (
      <div css={textStyles({ isSelected, isPlaceholder: false })}>
        <span>
          <b>
            {label}
            &nbsp;
          </b>
        </span>
        <span>{date}</span>
      </div>
    );
  }
  return (
    <span css={textStyles({ isSelected, isPlaceholder: true })}>
      {placeholder}
    </span>
  );
};

const Tabs = ({
  isSelectingStartDate,
  startSelectedLabel,
  endSelectedLabel,
  startPlaceholder,
  endPlaceholder,
  startDate,
  endDate,
  onTabClick,
  isDateRange,
  Icon
}) => (
  <div>
    <div css={calendarHeatherCntStyle()}>
      <button
        onClick={() => onTabClick({ isSelectingStartDate: true })}
        type="button"
        css={startTabStyles({ isSelectingStartDate })}
      >
        {renderTabContent(
          startDate,
          startSelectedLabel,
          startPlaceholder,
          isSelectingStartDate
        )}
        {Icon && <Icon css={iconStyles({ isSelectingStartDate })} />}
        <Triangle css={TriangleSvgStyles({ isSelectingStartDate })} />
      </button>
      {isDateRange && (
        <button
          onClick={() => onTabClick({ isSelectingStartDate: false })}
          type="button"
          css={endTabStyle({ isSelectingStartDate })}
        >
          {renderTabContent(
            endDate,
            endSelectedLabel,
            endPlaceholder,
            !isSelectingStartDate
          )}
        </button>
      )}
    </div>
  </div>
);

Tabs.propTypes = {
  isSelectingStartDate: PropTypes.bool,
  isDateRange: PropTypes.bool,
  startPlaceholder: PropTypes.string.isRequired,
  endPlaceholder: PropTypes.string.isRequired,
  startSelectedLabel: PropTypes.string.isRequired,
  endSelectedLabel: PropTypes.string.isRequired,
  startDate: PropTypes.string,
  endDate: PropTypes.string,
  onTabClick: PropTypes.func,
  Icon: PropTypes.func
};

Tabs.defaultProps = {
  isSelectingStartDate: true,
  isDateRange: true,
  startDate: null,
  endDate: null,
  onTabClick: () => {},
  Icon: null
};

export default Tabs;
