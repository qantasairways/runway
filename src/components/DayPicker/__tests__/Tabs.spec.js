import React from 'react';
import { shallow } from 'enzyme';
import Tabs from '../components/Tabs';

const CalendarIcon = () => <div>Calendar Icon</div>;

describe('Tabs', () => {
  it('renders correctly with defaults', () => {
    const calendar = shallow(
      <Tabs
        startPlaceholder="depart?"
        endPlaceholder="return?"
        startSelectedLabel="departing on"
        endSelectedLabel="returning on"
      />
    );

    expect(calendar).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    const calendar = shallow(
      <Tabs
        startPlaceholder="depart?"
        endPlaceholder="return?"
        startSelectedLabel="departing on"
        endSelectedLabel="returning on"
        isSelectingStartDate={false}
        startDate="6th June"
        endDate="10th October"
        Icon={CalendarIcon}
      />
    );

    expect(calendar).toMatchSnapshot();
  });

  it('renders correctly when isDateRange is false', () => {
    const calendar = shallow(
      <Tabs
        startPlaceholder="depart?"
        endPlaceholder="return?"
        startSelectedLabel="departing on"
        endSelectedLabel="returning on"
        isSelectingStartDate={false}
        startDate="6th June"
        endDate="10th October"
        Icon={CalendarIcon}
        isDateRange={false}
      />
    );

    expect(calendar).toMatchSnapshot();
  });
});
