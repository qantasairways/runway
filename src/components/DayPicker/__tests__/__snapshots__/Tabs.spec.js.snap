// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly when isDateRange is false 1`] = `
.emotion-4 {
  position: relative;
  width: 100%;
  background-color: #323232;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  font-family: "Ciutadella",sans-serif;
  position: relative;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 15px;
  border: solid 5px #323232;
  border-right: 0px;
  background-color: #323232;
  color: #fff;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: left;
}

.emotion-3:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-3 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-3::after {
  content: '';
  z-index: 5;
  position: absolute;
  right: -1px;
  border-top: 32px solid transparent;
  border-left: 0px;
  border-bottom: 32px solid transparent;
  border-right: 19px solid #fff;
  top: -5px;
}

@media only screen and (min-width:768px) {
  .emotion-3::after {
    right: -1px;
    border-left: 0px;
    border-top: 42px solid transparent;
    border-bottom: 42px solid transparent;
    top: -4px;
  }
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.25;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.625rem;
    max-width: 500px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media only screen and (min-width:960px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-1 {
  width: 28px;
  height: 28px;
  z-index: 10;
  position: absolute;
  right: -19px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

@media only screen and (min-width:321px) {
  .emotion-1 {
    width: 32px;
    height: 32px;
    right: -24px;
  }
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    width: 40px;
    height: 40px;
    right: -34px;
  }
}

.emotion-2 {
  z-index: 10;
  position: absolute;
  right: -12px;
  top: -7px;
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    height: 83px;
    right: -7px;
    top: -7px;
  }
}

<div>
  <div
    className="emotion-4"
  >
    <button
      className="emotion-3"
      tabIndex={-1}
      type="button"
    >
      <div
        className="emotion-0"
      >
        <span>
          <b>
            departing on
            Â 
          </b>
        </span>
        <span>
          6th June
        </span>
      </div>
      <CalendarIcon
        className="emotion-1"
      />
      <Triangle
        className="emotion-2"
      />
    </button>
  </div>
</div>
`;

exports[`Tabs renders correctly with defaults 1`] = `
.emotion-5 {
  position: relative;
  width: 100%;
  background-color: #323232;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  font-family: "Ciutadella",sans-serif;
  position: relative;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 15px;
  border: solid 5px #8de2e0;
  border-right: 0px;
  background-color: #fff;
  color: #323232;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: left;
}

.emotion-2:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-2 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-2::after {
  content: '';
  z-index: 5;
  position: absolute;
  right: -14px;
  border-top: 32px solid transparent;
  border-left: 15px solid #fff;
  border-bottom: 32px solid transparent;
  border-right: 0px;
  top: -5px;
}

@media only screen and (min-width:768px) {
  .emotion-2::after {
    right: -17px;
    border-left: 19px solid #fff;
    border-top: 42px solid transparent;
    border-bottom: 42px solid transparent;
    top: -4px;
  }
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.11;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.625rem;
    max-width: 500px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media only screen and (min-width:960px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-1 {
  z-index: 10;
  position: absolute;
  right: -48px;
  top: -5px;
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    height: 84px;
    right: -53px;
    top: -5px;
  }
}

.emotion-4 {
  font-family: "Ciutadella",sans-serif;
  position: relative;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  border: none;
  border-left: none;
  background-color: #323232;
  text-align: right;
  float: right;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0 15px;
}

.emotion-4:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-4 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-4 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media only screen and (min-width:768px) {
  .emotion-4 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<div>
  <div
    className="emotion-5"
  >
    <button
      className="emotion-2"
      tabIndex={-1}
      type="button"
    >
      <span
        className="emotion-0"
      >
        depart?
      </span>
      <Triangle
        className="emotion-1"
      />
    </button>
    <button
      className="emotion-4"
      tabIndex={-1}
      type="button"
    >
      <span
        className="emotion-0"
      >
        return?
      </span>
    </button>
  </div>
</div>
`;

exports[`Tabs renders correctly with props provided 1`] = `
.emotion-6 {
  position: relative;
  width: 100%;
  background-color: #323232;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-3 {
  font-family: "Ciutadella",sans-serif;
  position: relative;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 15px;
  border: solid 5px #323232;
  border-right: 0px;
  background-color: #323232;
  color: #fff;
  box-sizing: border-box;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: left;
}

.emotion-3:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-3 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-3::after {
  content: '';
  z-index: 5;
  position: absolute;
  right: -1px;
  border-top: 32px solid transparent;
  border-left: 0px;
  border-bottom: 32px solid transparent;
  border-right: 19px solid #fff;
  top: -5px;
}

@media only screen and (min-width:768px) {
  .emotion-3::after {
    right: -1px;
    border-left: 0px;
    border-top: 42px solid transparent;
    border-bottom: 42px solid transparent;
    top: -4px;
  }
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

.emotion-0 {
  font-size: 1.125rem;
  line-height: 1.25;
  font-weight: 400;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    font-size: 1.625rem;
    max-width: 500px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media only screen and (min-width:960px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-1 {
  width: 28px;
  height: 28px;
  z-index: 10;
  position: absolute;
  right: -19px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

@media only screen and (min-width:321px) {
  .emotion-1 {
    width: 32px;
    height: 32px;
    right: -24px;
  }
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    width: 40px;
    height: 40px;
    right: -34px;
  }
}

.emotion-2 {
  z-index: 10;
  position: absolute;
  right: -12px;
  top: -7px;
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

@media only screen and (min-width:768px) {
  .emotion-2 {
    height: 83px;
    right: -7px;
    top: -7px;
  }
}

.emotion-5 {
  font-family: "Ciutadella",sans-serif;
  position: relative;
  width: 50%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #323232;
  border: solid 5px #8de2e0;
  border-left: none;
  background-color: #fff;
  text-align: right;
  float: right;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 0 15px;
}

.emotion-5:focus {
  outline: none;
}

@media only screen and (min-width:768px) {
  .emotion-5 {
    height: 80px;
  }
}

@media only screen and (min-width:960px) {
  .emotion-5 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media only screen and (min-width:768px) {
  .emotion-5 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<div>
  <div
    className="emotion-6"
  >
    <button
      className="emotion-3"
      tabIndex={-1}
      type="button"
    >
      <div
        className="emotion-0"
      >
        <span>
          <b>
            departing on
            Â 
          </b>
        </span>
        <span>
          6th June
        </span>
      </div>
      <CalendarIcon
        className="emotion-1"
      />
      <Triangle
        className="emotion-2"
      />
    </button>
    <button
      className="emotion-5"
      tabIndex={-1}
      type="button"
    >
      <div
        className="emotion-0"
      >
        <span>
          <b>
            returning on
            Â 
          </b>
        </span>
        <span>
          10th October
        </span>
      </div>
    </button>
  </div>
</div>
`;
